edx_exams/apps/core/signals/signals.py

```diff
@@ -0,0 +1,33 @@
+"""
+Signal definitions and functions to send those signals for the edx-exams application.
+"""
+
+from openedx_events.learning.data import ExamAttemptData, UserData, UserPersonalData
+from openedx_events.learning.signals import EXAM_ATTEMPT_SUBMITTED
+
+
+def emit_exam_attempt_submitted_event(user, course_key, usage_key, exam_type):
+    """
+    Emit the EXAM_ATTEMPT_SUBMITTED Open edX event.
+    """
+    name = user.full_name or ''
```

Comment:
This is because the `full_name` field of the `User` model can either be the empty string (`blank=True`) or None (`null=True`). This comes from the edX [cookiecutter](https://github.com/openedx/edx-cookiecutters/blob/df10fdadc365f41e81bf2429c8b68ab13be4cb3c/cookiecutter-django-ida/%7B%7Bcookiecutter.repo_name%7D%7D/%7B%7Bcookiecutter.project_name%7D%7D/apps/core/models.py#L17). I'm not sure what the rationale was for designing the model this way; setting both attributes to true is something [Django discourages](https://docs.djangoproject.com/en/4.2/ref/models/fields/#null). I created a thread in Slack to try to get some information on this decision. For now, I've done this. I'd prefer to fix the `User` model, but this will do until I have more information. I'll discuss this during standup.